lib/sqlalchemy/cyextension/resultproxy.pyx

```diff
@@ -67,28 +57,17 @@ cdef class BaseRow:
     def __getitem__(self, index):
         return self._data[index]
 
-    cpdef _get_by_key_impl_mapping(self, key):
-        try:
-            rec = self._keymap[key]
-        except KeyError as ke:
-            rec = self._parent._key_fallback(key, ke)
-
-        mdindex = rec[MD_INDEX]
-        if mdindex is None:
-            self._parent._raise_for_ambiguous_column_name(rec)
-        elif (
-            self._key_style == _KEY_OBJECTS_ONLY
-            and isinstance(key, int)
-        ):
-            raise KeyError(key)
+    def _get_by_key_impl_mapping(self, key):
+        return self._get_by_key_impl(key, 0)
 
-        return self._data[mdindex]
+    cdef _get_by_key_impl(self, object key, int attr_err):
+        index = self._key_to_index.get(key)
+        if index is not None:
+            return self._data[<int>index]
```

Comment:
```suggestion
            return self._data[index]
```

I think we can drop the `<int>` conversion in the cython

<img width="1439" alt="Screenshot 2023-04-19 at 12 10 53 PM" src="https://user-images.githubusercontent.com/663432/233210969-c2cf6292-a414-486c-911c-6b7a849d9607.png">

as it means we have to __Pyx_PyInt_As_int to convert it back to the python int since __Pyx_GetItemInt_Tuple expects a python int

<img width="1411" alt="Screenshot 2023-04-19 at 12 10 57 PM" src="https://user-images.githubusercontent.com/663432/233210981-01a1e292-b124-4e7d-8cb2-3efd3070ae5e.png">

Reply:
testing that now to see if its faster or slower because `Pyx_GetItemInt_Tuple` might still be better even with another conversion

Reply:
`<int>` does seem faster.

I'm wondering if we can avoid the `PyInt_as_int` but I guess not

Reply:
yes, but I wanted the pyx_getitem_tuple instead of the pyx_pyobject_getitem

not sure if there is a way of making it use that without the cast

Reply:
doing `self._data[int(index)]` make it use `Pyx_GetItemInt_Tuple` but it seems slower

Reply:
The int conversion effective boils it down to `PyTuple_GET_ITEM` with `cpython`

Reply:
`int` seems better 👍

Reply:
I don't see anything else we can squeeze out of this

Reply:
indeed. I don't know if there is a way of making it do "cast to python int" instead of a c-int.

Reply:
> I don't see anything else we can squeeze out of this

yes, whatever you do it seems to have no effect or make things worse.

Overall I think conceptually there isn't anything left to optimize since we basically do:
- get the index from the dict
- if we find an index return the thing from the tuple
- otherwise raise

not sure if using a linear probe in case of a small row is worth trying

Reply:
Was looking at the Py_None checks but it looks like they don't matter https://github.com/cython/cython/issues/3797